### PR TITLE
Pose graph: do not mark all submaps finished when finalizing a trajectory

### PR DESCRIPTION
**Pull request by ojura**
I think this was a mistake in #563 (@jihoonl). Not all submaps should be marked as finished, only those pertaining to the trajectory being finalized.

### CODE REVIEW

File: cartographer/mapping_2d/pose_graph.cc [hunk 1]
```diff
@@ -381,9 +381,8 @@ void PoseGraph::FinishTrajectory(const int trajectory_id) {
     CHECK_EQ(finished_trajectories_.count(trajectory_id), 0);
     finished_trajectories_.insert(trajectory_id);
 
-    auto submap_data = optimization_problem_.submap_data();
-    for (auto submap_id_data : submap_data) {
-      submap_data_.at(submap_id_data.id).state = SubmapState::kFinished;
+    for (const auto& submap : submap_data_.trajectory(trajectory_id)) {
+      submap_data_.at(submap.id).state = SubmapState::kFinished;
```
Comment (ojura) [inline cartographer/mapping_2d/pose_graph.cc]: This could look nicer if `MapById` provided a non-const iterator as well.

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 1.

**Comment by jihoonl** [comment]
Ah good catch. Thanks

**Pull request merged**